lib/validation/validator.rb

```diff
@@ -1,132 +1,172 @@
 module Validation
-  module Rules
-    # A hash of rules for this object
-    def rules
-      @rules ||= {}
-    end
+  module Validator
+    attr_reader :object, :results
 
-    # A hash of errors for this object
-    def errors
-      @errors ||= {}
+    def initialize(object)
+      raise ArgumentError, "cannot validate nil" if object.nil?
```

Comment:
> and had a way to define rules that validate the object itself, and not one of its fields

Well, `Validation` works on any object that responds to messages (any object, basically). You *can* validate some things of `nil` (since it's an object that responds to messages), it probably just won't be useful very often.

This means you *could* validate nearly any "primitive" value like `Integer`, etc, you'd just need to validate it's method calls.